.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,287 @@
+        with:
+          name: build-files
+          path: |
+            rskj-core/build
+
+      - name: If GH event is PR
+        if: github.event_name == 'pull_request'
+        run: |
+          git branch -f master origin/master
+          pr_number==${{ github.event.pull_request.number }}
+          extra_flags="-Dsonar.pullrequest.base=master
+            -Dsonar.pullrequest.branch=${{ github.head_ref }}
+            -Dsonar.pullrequest.key=$pr_number"
+          ./gradlew sonarqube --no-daemon -x build -x test \
+            $extra_flags \
+            -Dsonar.organization=rsksmart \
+            -Dsonar.host.url="https://sonarcloud.io" \
+            -Dsonar.token="${{ secrets.SONAR_TOKEN }}"
+
+      - name: If not
+        if: github.event_name != 'pull_request'
+        run: |
+          ./gradlew sonarqube --no-daemon -x build -x test \
+            -Dsonar.branch.name=master \
+            -Dsonar.organization=rsksmart \
+            -Dsonar.host.url="https://sonarcloud.io" \
+            -Dsonar.token="${{ secrets.SONAR_TOKEN }}"
+
+  unit-tests:
+    needs: prepare-rskj
+    runs-on: ubuntu-latest
+    container:
+      image: openjdk:8-jdk
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup System Tools
+        run: |
+          apt update -y
+          apt install -y curl
+
+      - uses: actions/cache@v4
+        name: Restore gradle cache
+        id: cache-gradle
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get gradle wrapper and build dependencies
+        run: |
+          if [ ! -f gradle/wrapper/gradle-wrapper.jar ]; then
+            ./configure.sh
+            ./gradlew --no-daemon dependencies
+          fi
+
+      - name: Run tests
+        run: |
+          ./gradlew --no-daemon --stacktrace test
+
+  integration-tests:
+    needs: prepare-rskj
+    runs-on: ubuntu-latest
+    container:
+      image: openjdk:8-jdk
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup System Tools
+        run: |
+          apt update -y
+          apt install -y curl
+
+      - uses: actions/cache@v4
+        name: Restore gradle cache
+        id: cache-gradle
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get gradle wrapper and build dependencies
+        run: |
+          if [ ! -f gradle/wrapper/gradle-wrapper.jar ]; then
+            ./configure.sh
+            ./gradlew --no-daemon dependencies
+          fi
+
+      - name: Run tests
+        run: |
+          ./gradlew --no-daemon --stacktrace integrationTest
+
+  mining-tests:
```

Comment:
yea that's how it is done in circle, but I wanted to avoid that, since I didn't see a compelling reason to run an action in repo A to trigger an action in repo B that clones repo A at the branch of the PR.
If it is done that way to allow other repos to use it as well, then we should convert that repo to a github action and run it in their workflow like
```yaml
jobs:
  mit:
    steps:
      - name: mit
        uses: rsksmart/mining-integration-tests
```
shouldn't we?